### PR TITLE
[#8] Add Kafka receiver to CLI.

### PR DESCRIPTION
**Pull request by b-abel**
This PR adds a receiver to the CLI that consumes telemetry data and events from Kafka. Please refer to "Start the Downstream Consumer" in https://github.com/eclipse/hono/pull/2216#issuecomment-730255279 for usage instructions.

This belongs to #8.

### CODE REVIEW

File: cli/src/main/java/org/eclipse/hono/cli/app/KafkaReceiver.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+     */
+    @PostConstruct
+    Future<Void> start() {
+        return createConsumer()
+                .onComplete(this::handleCreateConsumerStatus);
+    }
+
+    private Future<Void> createConsumer() {
+
+        if (tenantId == null || tenantId.isEmpty()) {
```
Comment (calohmn) [inline cli/src/main/java/org/eclipse/hono/cli/app/KafkaReceiver.java:87]: Shorter: `Strings.isNullOrEmpty(tenantId)`

File: cli/src/main/java/org/eclipse/hono/cli/app/KafkaReceiver.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            log.info("received message [topic: {}, key: {}]: {}", record.topic(), record.key(), record.value());
+        }
+
+    }
+
+    private void logVerbosely(final KafkaConsumerRecord<String, Buffer> record) {
+        final long timestamp = record.timestamp();
+        final LocalDateTime time = LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.systemDefault());
+
+        final StringBuffer buffer = new StringBuffer();
```
Comment (calohmn) [inline cli/src/main/java/org/eclipse/hono/cli/app/KafkaReceiver.java:149]: Better use StringBuilder.

**Comment by b-abel** [comment]
I added the requested changes.

**Comment by calohmn** [comment]
When looking at the non-verbose output
````
received message [topic: hono.telemetry.DEFAULT_TENANT, key: 4711]: {"temp": 6}
````
I think it could maybe be less Kafka-generic and more Hono-specific, mimicking the AMQP output and picking the relevant parts of the Kafka record:
````
received [telemetry|event] message [device: {}, content-type: {}]: payload
````
WDYT?

**Comment by b-abel** [comment]
> I think it could maybe be less Kafka-generic and more Hono-specific, mimicking the AMQP output and picking the relevant parts of the Kafka record:

Done. Finding the endpoint name and getting the content type looks not very nice ATM, but I would leave that for now and add that logic to a central place with the next PR for #2392.

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**